File: php/list_servers.php
<?php
/**
 * List all servers with the DataMammoth PHP SDK.
 *
 * Usage:
 *   export DM_API_KEY="dm_live_..."
 *   php list_servers.php
 */

require_once __DIR__ . '/../../vendor/autoload.php';
// Or if installed via composer: require_once 'vendor/autoload.php';

use DataMammoth\DataMammoth;

$apiKey = getenv('DM_API_KEY');
if (!$apiKey) {
    fwrite(STDERR, "Error: Set the DM_API_KEY environment variable\n");
    exit(1);
}

$dm = new DataMammoth($apiKey);

// List servers with filters
$response = $dm->servers->list([
    'status' => 'running',
    'sort' => '-created_at',
    'per_page' => 20,
]);

$servers = $response['data'] ?? [];
$meta = $response['meta'] ?? [];
$pagination = $meta['pagination'] ?? $meta;

$page = $pagination['page'] ?? 1;
$totalPages = $pagination['total_pages'] ?? 1;
$total = $pagination['total'] ?? count($servers);

echo "Servers (page {$page}/{$totalPages}, total: {$total}):\n\n";

foreach ($servers as $server) {
    $icon = match ($server['status'] ?? '') {
        'running' => '[OK]',
        'stopped' => '[--]',
        'provisioning' => '[..]',
        'terminated' => '[XX]',
        default => '[??]',
    };

    echo "  {$icon} {$server['id']}\n";
    echo "      Hostname: " . ($server['hostname'] ?? '(none)') . "\n";
    echo "      IP:       " . ($server['ip_address'] ?? 'pending') . "\n";
    echo "      Region:   " . ($server['region'] ?? '') . "\n";
    echo "      Plan:     " . ($server['plan'] ?? '') . "\n\n";
}

// Auto-paginate through all servers
echo "--- All servers (all pages) ---\n";
foreach ($dm->servers->listAll(['status' => 'running']) as $server) {
    $id = $server['id'];
    $hostname = $server['hostname'] ?? '(none)';
    $status = $server['status'] ?? '';
    echo "  {$id}: {$hostname} ({$status})\n";
}

File: go/list_servers.go
// List all servers with the DataMammoth Go SDK.
//
// Usage:
//
//	export DM_API_KEY="dm_live_..."
//	go run list_servers.go
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	datamammoth "github.com/datamammoth/dm-go"
)

func main() {
	apiKey := os.Getenv("DM_API_KEY")
	if apiKey == "" {
		log.Fatal("Error: Set the DM_API_KEY environment variable")
	}

	client := datamammoth.NewClient(apiKey)
	ctx := context.Background()

	// List servers with filters
	servers, meta, err := client.Servers.List(ctx, &datamammoth.ListOptions{
		Status:  "running",
		Sort:    "-created_at",
		PerPage: 20,
	})
	if err != nil {
		log.Fatalf("Failed to list servers: %v", err)
	}

	fmt.Printf("Servers (page %d/%d, total: %d):\n\n", meta.Page, meta.TotalPages, meta.Total)

	for _, s := range servers {
		icon := "[??]"
		switch s.Status {
		case "running":
			icon = "[OK]"
		case "stopped":
			icon = "[--]"
		case "provisioning":
			icon = "[..]"
		case "terminated":
			icon = "[XX]"
		}

		fmt.Printf("  %s %s\n", icon, s.ID)
		fmt.Printf("      Hostname: %s\n", s.Hostname)
		fmt.Printf("      IP:       %s\n", s.IPAddress)
		fmt.Printf("      Region:   %s\n", s.Region)
		fmt.Printf("      Plan:     %s\n\n", s.Plan)
	}
}
